tests/IVIMmodels/unit_tests/test_ivim_fit.py

```diff
@@ -152,6 +156,83 @@ def test_bounds(bound_input, eng):
+        else:
+            kwargs = {'eng': eng}
+    else:
+        kwargs={}
+    fit = OsipiBase(algorithm=algorithm,**kwargs)
+    if hasattr(fit, 'ivim_fit_full_volume') and callable(getattr(fit, 'ivim_fit_full_volume')):
+        start_time = time.time()  # Record the start time
+        fit_result = fit.osipi_fit_full_volume(data, bvals)
+        elapsed_time2 = time.time() - start_time  # Calculate elapsed time
+        print('time elaapsed is '+str(elapsed_time2))
+        assert np.shape(fit_result['D'])[0] == np.shape(data)[0]
+        assert np.shape(fit_result['D'])[1] == np.shape(data)[1]
+        assert np.shape(fit_result['D'])[2] == np.shape(data)[2]
+        # check if right variable is in right place
+        assert np.nanmean(fit_result['D']) < np.nanmean(fit_result['Dp'])
+        assert np.nanmean(fit_result['Dp']) < np.nanmean(fit_result['f'])
+    else:
+        pytest.skip(reason="Wrapper has no ivim_fit_full_volume option")
+
+
+def test_parallel(algorithmlist,eng,threeddata):
+    algorithm, kwargs = algorithmlist
+    data, Dim, fim, Dpim, bvals = threeddata
+    # Get index of b=0
+    b0_index = np.where(bvals == 0)[0][0]
+    data[data < 0] = 0
+    # Mask of voxels where signal at b=0 >= 0.5
+    invalid_mask = data[:, :, :, b0_index] < 0.01
+    data[invalid_mask,:] = np.nan
+    print('testing ' + str(np.sum(~invalid_mask)) + ' voxels of a matrix size ' + str(np.shape(data)))
+    if kwargs.get("requires_matlab", False):
+        if eng is None:
+            pytest.skip(reason="Running without matlab; if Matlab is available please run pytest --withmatlab")
+        else:
+            kwargs = {'eng': eng}
+    else:
+        kwargs={}
+    fit = OsipiBase(algorithm=algorithm,**kwargs)
+
+    def dummy_task(x):
+        return x
+
+    start_time = time.time()  # Record the start time
+    fit_result = fit.osipi_fit(data, bvals,njobs=1)
+    time_serial = time.time() - start_time  # Calculate elapsed time
+    Parallel(n_jobs=2)(delayed(dummy_task)(i) for i in range(8)) #github actions only supports 2 cores
+    start_time = time.time()  # Record the start time
+    fit_result2 = fit.osipi_fit(data, bvals,njobs=2)
+    time_parallel= time.time() - start_time  # Calculate elapsed time
+
+    print('singular took '+str(time_serial)+' seconds, and parallel took '+str(time_parallel)+' seconds')
+    assert np.shape(fit_result['D'])[0] == np.shape(data)[0]
+    assert np.shape(fit_result['D'])[1] == np.shape(data)[1]
+    assert np.shape(fit_result['D'])[2] == np.shape(data)[2]
+    if not fit.stochastic:
+        assert np.allclose(fit_result['D'], fit_result2['D'], atol=1e-4), "Results differ between parallel and serial"
+        assert np.allclose(fit_result['f'], fit_result2['f'], atol=1e-2), "Results differ between parallel and serial"
+        assert np.allclose(fit_result['Dp'], fit_result2['Dp'], atol=1e-2), "Results differ between parallel and serial"
+    if time_parallel * 1.3 > time_serial:
```

Comment:
Time can be tricky because it can vary based on system load. Probably a good choice that's it's not a requirement but this message probably won't ever get seen. I wonder if there's some report we could write to - perhaps in the future - for algorithm profiling purposes?